featuretools/primitives/base/primitive_base.py

```diff
@@ -98,3 +105,33 @@ def get_arguments(self):
             values.append((name, value))
 
         return values
+
+    def get_description(self, input_column_descriptions, slice_num=None, template_override=None):
+        template = template_override or self.description_template
+        if template:
+            if isinstance(template, list):
+                if slice_num is not None:
+                    slice_index = slice_num + 1
+                    if slice_index < len(template):
+                        return template[slice_index].format(*input_column_descriptions,
+                                                            nth_slice=convert_to_nth(slice_index))
+                    else:
+                        if len(template) > 2:
+                            raise IndexError('Slice out of range of template')
+                        return template[1].format(*input_column_descriptions,
+                                                  nth_slice=convert_to_nth(slice_index))
+                else:
+                    template = template[0]
+            return template.format(*input_column_descriptions)
+
+        # generic case:
+        name = self.name.upper() if self.name is not None else type(self).__name__
```

Comment:
do we have a test case for the scenario where the primitive is missing the name attribute?